### PR TITLE
Added new tutorial on PRNGs with RandomVariables

### LINKED ISSUE #12: DOC: Extend documentation on RandomVariables
### Idea or request for content:

I currently wrote a Gist just to investigate how updates work, that I think would be nice to add to the official docs. It needs to be extended slightly on the Scan part, and perhaps also add one on cloning graphs and the issue with default_updates

https://gist.github.com/ricardoV94/7aa6c23726f20fbec95a1cda449c15f5

### PR DESCRIPTION
**Pull request by HangenYuu**
## Description

Added new tutorial on PRNGs with RandomVariables per @ricardoV94 suggestions.

## Related Issue

- [ ] Closes #12 
- [ ] Related to #

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [x] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

**Review by HangenYuu — COMMENTED** [review]
I also realized some minor erros in contributing documentation and dependencies need updating in `environment.yml` file. Will open a separate PR for it.

File: doc/tutorial/prng.rst [hunk 1]
```diff
@@ -0,0 +1,720 @@
...
+>>> rng = pytensor.shared(np.random.default_rng(123))
+>>> next_rng, x = pt.random.uniform(rng=rng).owner.outputs
+>>> 
+>>> rng.default_update = next_rng
+>>> f = pytensor.function([], x)
+>>> 
+>>> f(), f(), f()
+(array(0.68235186), array(0.05382102), array(0.22035987))
+
+Which is exactly what RandomStream does behind the scenes
```
Comment (ricardoV94) [inline doc/tutorial/prng.rst:217]: Emphasize a bit more the switch to `RandomStream` and link to the respective docs. We mention at the top that there are two API: numpy-like -> RandomStream, scipy-like -> pt.random.foo. We should re-emphasize that when we mention RandomStreams here.

File: doc/tutorial/prng.rst [hunk 2]
```diff
@@ -0,0 +1,720 @@
...
+>>> x = pt.random.uniform(rng=rng_shared, name="x")
+>>> f = pytensor.function([], x)
+>>> pytensor.dprint(f, print_destroy_map=True) # doctest: +SKIP
+uniform_rv{0, (0, 0), floatX, False}.1 [id A] 'x' 0
+ |rng [id B]
+ |TensorConstant{[]} [id C]
+ |TensorConstant{11} [id D]
+ |TensorConstant{0.0} [id E]
+ |TensorConstant{1.0} [id F]
+<ipykernel.iostream.OutStream at 0x7f6c308c3eb0>
```
Comment (ricardoV94) [inline doc/tutorial/prng.rst:287]: Update all dprints to show how it looks today, now that we have less inputs

**Comment by ricardoV94** [comment]
Looks good, but the Scipy snippet is not formatting correctly: https://pytensor--928.org.readthedocs.build/en/928/tutorial/prng.html#scipy

File: doc/tutorial/index.rst [hunk 3]
```diff
@@ -59,6 +59,7 @@ Further reading
 .. toctree::
 
     loading_and_saving
+    prng
```
Comment (ricardoV94) [inline doc/tutorial/index.rst:62]: Should be after sparse, in the advanced section above?

Comment (HangenYuu) [reply doc/tutorial/index.rst]: I changed the minor issues.

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
Thanks @HangenYuu !

**Comment by HangenYuu** [comment]
> Thanks @HangenYuu !

No problem @ricardoV94!

**Pull request merged**